src/utf8.rs

```diff
@@ -132,11 +132,13 @@ pub fn decode_last_utf8(src: &[u8]) -> Option<(char, usize)> {
             break;
         }
     }
-    match decode_utf8(&src[start..]) {
-        None => None,
-        Some((_, n)) if n < src.len() - start => None,
-        Some((cp, n)) => Some((cp, n)),
-    }
+    decode_utf8(&src[start..]).and_then(|(cp, n)| {
+        if n < src.len() - start {
+            None
+        } else {
+            Some((cp, n))
+        }
+    })
```

Comment:
I prefer the previous version. Here, you have to process a monadic bind followed by case analysis. Before, it was just case analysis.